### PR TITLE
Add some TypeBuilder tests

### PR DESCRIPTION
**Pull request by hughbe**
And consolidate some others

/cc @stephentoub

### CODE REVIEW

**Comment by karelz** [comment]
@AtsushiKan, @stephentoub can you please review the change? (or nominate someone else ;-))

File: src/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderCreateType.cs [hunk 1]
```diff
@@ -29,6 +29,8 @@ public void CreateType(TypeAttributes attributes)
             TypeBuilder type = Helpers.DynamicType(attributes);
             Type createdType = type.CreateTypeInfo().AsType();
             Assert.Equal(type.Name, createdType.Name);
+
+            Assert.Same(type.CreateTypeInfo(), type.CreateTypeInfo());
```
Comment (hughbe) [inline src/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderCreateType.cs]: Thanks for letting me know. I've fixed this, and all other occurences of `Assert.Same` to `Assert.Equal`

**Comment by hughbe** [comment]
I've had to rebase the branch due to merge conflicts. Sorry!

**Comment by AtsushiKan** [comment]
CI is failing with some relevant looking build errors.

862>Utilities.cs(108,26): error CS1503: Argument 1: cannot convert from 'System.Reflection.Emit.TypeBuilder' to 'string' [D:\j\workspace\windows_nt_de---4526f5ff\src\System.Reflection.Emit\tests\System.Reflection.Emit.Tests.csproj]
   862>Utilities.cs(108,32): error CS1503: Argument 2: cannot convert from 'System.Type' to 'string' [D:\j\workspace\windows_nt_de---4526f5ff\src\System.Reflection.Emit\tests\System.Reflection.Emit.Tests.csproj]
   862>TypeBuilder\TypeBuilderDefineField.cs(44,26): error CS1503: Argument 1: cannot convert from 'System.Reflection.Emit.TypeBuilder' to 'string' [D:\j\workspace\windows_nt_de---4526f5ff\src\System.Reflection.Emit\tests\System.Reflection.Emit.Tests.csproj]
   862>TypeBuilder\TypeBuilderDefineField.cs(44,32): error CS1503: Argument 2: cannot convert from 'System.Type' to 'string' [D:\j\workspace\windows_nt_de---4526f5ff\src\System.Reflection.Emit\tests\System.Reflection.Emit.Tests.csproj]
   862>TypeBuilder\TypeBuilderDefineGenericParameters.cs(37,30): error CS1503: Argument 1: cannot convert from 'System.Reflection.Emit.TypeBuilder' to 'string' [D:\j\workspace\windows_nt_de---4526f5ff\src\System.Reflection.Emit\tests\System.Reflection.Emit.Tests.csproj]
   862>TypeBuilder\TypeBuilderDefineGenericParameters.cs(37,36): error CS1503: Argument 2: cannot convert from 'System.Type' to 'string' [D:\j\workspace\windows_nt_de---4526f5ff\src\System.Reflection.Emit\tests\System.Reflection.Emit.Tests.csproj]
   862>Done Building Project "D:\j\workspace\windows_nt_de---4526f5ff\src\System.Reflection.Emit\tests\System.Reflection.Emit.Tests.csproj" (Build target(s)) -- FAILED.

**Comment by hughbe** [comment]
Yup, got lazy and didn't build after the Assert.Same -> Assert.Equal change! Turns out `TypeBuilder` only inherits from `Type` in the full framework ugh

**Pull request merged**